Sources/Preferences/Preferenceable.swift

```diff
@@ -1,6 +1,38 @@
 import Cocoa
 
+public struct PreferenceIdentifier: Equatable, RawRepresentable {
+	public let rawValue: String
+
+	public init(rawValue: String) {
+		self.rawValue = rawValue
+	}
+}
+
 public protocol Preferenceable: AnyObject {
+	var preferenceIdentifier: PreferenceIdentifier { get }
 	var toolbarItemTitle: String { get }
-	var toolbarItemIcon: NSImage { get }
+	var toolbarItemIcon: NSImage? { get }
+	var viewController: NSViewController { get }
+}
```

Comment:
Currently, if you specify `.segmentedControl` as style, you still have to implement `toolbarItemIcon`, which is weird. I think we should make it truly optional, but adding:

```swift
extension Preferenceable {
	public var toolbarItemIcon: NSImage? {
		return nil
	}
}
```

Reply:
Actually, we would have to make it:

```swift
public protocol Preferenceable: AnyObject {
	var preferenceIdentifier: PreferenceIdentifier { get }
	var toolbarItemTitle: String { get }
	var toolbarItemIcon: NSImage { get } // not optional
	var viewController: NSViewController { get }
}

extension Preferenceable {
	public var toolbarItemIcon: NSImage {
		return NSImage(size: .zero)
	}
}
```

That way the user can either specify a non-optional ` let toolbarItemIcon = NSImage(named: NSImage.preferencesGeneralName)!` or nothing at all.

Reply:
sounds good 👍